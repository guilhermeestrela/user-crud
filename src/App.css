.App {
  text-align: center;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

.container {
  background-color: #fff;
  border-radius: 5px;
  padding: 16px;
  width: 80%;
}
