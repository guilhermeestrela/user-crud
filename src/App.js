import React, {useEffect, useState} from 'react'
import { Form } from './components/Form'
import { UserList } from './components/UserList';
import './App.css';
import { getUsers, createUser, deleteUser } from './services/api'

function App() {
  const [ users, setUsers ] = useState([])

  const getApiUsers = async () => {
      const userList = await getUsers()
      const tmpArray = []
      for (let key in userList) {
          tmpArray.push(userList[key])
      }
      setUsers(tmpArray)
  }

  const addUser = async (user) => {
    const result = await createUser(user)
    if (result.status === 'ok') {
      getApiUsers()
    }
  }

  const removeUser = async (id) => {
    const result = await deleteUser(id)
    if (result.status === 'ok') {
      getApiUsers()
    }
  }

  useEffect(() => {
     getApiUsers()
    }, [])
    
  return (
    <div className="App">
      <div className="container">
        <Form addUser={addUser} />
        <UserList users={users} removeUser={removeUser}/>
      </div>
    </div>
  );
}

export default App;
