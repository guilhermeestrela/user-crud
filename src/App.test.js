import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'

import App from './App'

describe('App', () => {
    beforeEach(() => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => Promise.resolve({ 
                    1637098364443: {
                        id: 1637098364443,
                        name: "Test User"
                    } 
                }),
            })
        );
    })

    it('Should render App correctly', async () => {
        const { container, getByText } = render(<App />)

        await waitFor(() => getByText('Test User'))

        expect(container).toMatchSnapshot()
        expect(getByText('Test User')).toBeInTheDocument()
    })
})
