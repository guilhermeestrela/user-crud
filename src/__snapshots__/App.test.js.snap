// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render App correctly 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <form>
        <input
          class="input"
          placeholder="Input glover's name"
          type="text"
          value=""
        />
        <button
          class="primary-button"
          type="submit"
        >
          ADD
        </button>
      </form>
      <ul
        class="user-list"
      >
        <li
          class="user-item"
        >
          <p
            class="user-name"
          >
            Test User
          </p>
          <button
            class="secondary-button"
          >
            Remove
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
